### ISSUE TITLE
suggestions

### ISSUE THREAD

**Issue by Papa2Cherries** [tabwrangler/tabwrangler#192]
1) add the sorting of websites via site url, not just the site title.  That way when you sort it all the imdb.com are next to one another, all of the youtube.com's are next to one another, etc.  

2) add a X button to delete a webpage, this will make management a lot faster.  Not a submenu like right click, but an actual button that appears when hovering over.  or at the very least when you hold down a button like shift, or control, that way you can manage old tabs not needed MUCH faster.  This feature is in almost all the other tab manager programs, this would be a great addition.

3) allow the top utility area to follow when scrolling down.  Especially the search field.  that way you don't have to scroll all the way back up to modify your search parameters or interact with any other button currently at the top.

4)  allow apple-f (command-f) to populate the search field for real quick searching instead of scrolling to the top manually either via scroll wheel or clicking on the bar.  this would be lessened with the suggestion #3, but it still would be also more helpful and fluid if you added this feature.  

5)  with the slowness of removing tabs no longer needed, either opening them in a tab then closing that tab (slow) or manually clicking on the small box, then scrolling to the top and clicking deleted checked (almost worse), one way that may allow for another smoother way to manage would be the ability to highlight them like you would in finder, with perhaps shift or command/control, forming the group of ones you don't want then you can delete them with the button.  Even at this point adding a right click menu item of "remove all tabs selected?" would be helpful so not needing to scroll to the top.  On top of this suggestion, a slightly different way to do this, would be to allow arrow scrolling up and down, and allowing space bar to check/uncheck the boxes, once again to allow faster managing.  we are wrangling tabs here, and that can get into the 100's of tabs, so anything that cuts down movements and seconds, helps heaps.  

I would love to say, thank you for this addon.  it is great.  Fantastic way to manage.  I hope you will look into adding this to safari and firefox.  Especially safari, it does not have anything even remotely good enough.  Firefox has decent alternatives, but man safari is lacking, probably due to having to get official certificate via apple developer, but hey, just make it github and we can install in a round about way.  Please please add these ideas as well as make a safari counterpart.  This way of managing tabs is vastly superior to others at the moment.

**Comment by Papa2Cherries** [comment]
6) just thought of another one.  You have the option of adding a single second for the window to be open for it to count as a reset.  Perhaps allow a box that we can fill out ourselves for how many seconds we want it to be open to count.  I often scroll through pages left and right via a next tab/back tab function, and am looking for a tab, often quickly, but that one second counts sometimes as it is too short.  if adding a box is too hard, perhaps add a 1, 3, 5, 15, 30, 60 second option.

**Comment by Papa2Cherries** [comment]
ok i just realized my #3 above somehow hyperlinked to some unintentional post.  I meant my 3rd suggestion :P

**Comment by Papa2Cherries** [comment]
I would try to help with the safari port, not that i presently know how to do this, but am earnest and know the basics of coding from when i was in college, that i would take the time to work on it and learn, just need help or a good reference on how to start then i could collaborate.

**Comment by ssorallen** [comment]
Thank you for the thorough issue and the time you spent writing it up. I appreciate any feedback and help with this extension.

1. Great idea, I opened #193 for this.
2. Making tabs deletable with a single click has been requested before. Let's revive it for real. I re-opened #145 for this.
3. The buttons for multi-select / delete / restore and sort should scroll with you when you scroll in Tab Wrangler's view. Does that not happen for you? It would be nice to add the search bar in there too.
4. Cmd+f is owned by Chrome itself for "find in page", but we could add another shortcut. "/" is a common shortcut for focusing search; it's used in Gmail. Would you use that?
5. Multi-select should already work! Hold shift when clicking, and all checkboxes between should be toggled.
6. There is an option called "Reset a tab's timer only after it is active for 1 second". Does that achieve what you would like?

**Comment by ssorallen** [comment]
I have never developed an extension for Safari, but I will open a task to explore it.

**Comment by quintrino** [comment]
I like the '/' keyboard shortcut for focusing search.

**Comment by ssorallen** [comment]
All of the suggestions are now captured in other issues (or already implemented!). I opened #205 to capture the search focus. I'm going to close this.

Feel free to comment and/or re-open if you have other questions.

**Comment by Papa2Cherries** [comment]
Apologies, saw your questions and forgot to respond.  Ironically, i think the tab i opened to respond to this got wrangled.  Out of sight, out of mind....  

re 6  readdressing the number 6 suggestion. Anywho, the option for allow a window to be open for 1 second was what i originally wrote about with # 6 suggestion.  I was implying that 1 second is not long enough.  Allow us to have more check boxes or a open field that we can input the seconds we want for the threshold. so either something like check box options of 1 3 5 15 30 60 seconds, or just allow us to write in what we want.  

7, I am here to report a very dangerous bug.  I started deleting various tabs out of my wrangled area and noticed that on more than one occasion, about every 4 to 7 clicks, the tab i chose to delete is not the one that got deleted, this also happens for tabs i click to reopen.  What seems to be happening, is i click delete or click on link to reopen, and a DIFFERENT tab gets deleted from the list.  the list count goes down by one, the clicked link effectively is no longer under your mouse but one row higher.  example:  link "how a tv works in slow motion" is under my mouse.  I left click the link, and that correct link is placed into a tab.  however, the link remains in the wrangled list only one row higher.  yet the list is diminished by one.  I have not figured out how it decides which link is removed, but it is removing them.  When i was in time descending, it removed the top 4 links.  I did a search for them, and they are not present.  to put it plainly, the link you are manipulating and the link the program decides to remove from the wrangled list is not always the same.  PLEASE FIX ASAP>>>PROGRAM UNUSABLE until then.

**Comment by ssorallen** [comment]
@Papa2Cherries Thanks for the details. I captured the two issues you mentioned in #215 and #216. I have an idea what might cause #216, but I will need to investigate. If you are able to reproduce it consistently, please comment in that issue so I can reproduce it locally on my computer.

**Comment by Papa2Cherries** [comment]
# 8, i am having an issue causing all my tabs to close now that 6.1.1 is out, im confidently and boldy allowing the extension to ride.  when i set the timer to something low like 30 seconds, it appeared to be working quite well, went from 115 to 483 rather quickly.  But sadly it is staying there.  I have at least a couple hundred more.  and my only close if more than is set to 20, so that is not the reason.  I watch the timer go down, and it just goes 3  2  1  --, that last part being the timer stopped with two dashes.  They are not being wrangled.  I am not exactly at the old 500 limit, nor am i at the new 1000 limit.  I have tried to get it to take them all again and again, but it seems that they are just counting down and then nothing happens. SO this is a a bug/suggestion report.  Thought it fitting to keep all mine in the same post for easy to follow up on.  Anywho, is it possible to add a right click or something, that says "wrangle all tabs in this window" or "all other tabs not in this window" etc.  Im sure you can come up with a few useful options.  I am thinking about this because the above problem is not auto corralling them, but i can manually, 1 at a time, do it.  which is horrendous.  actually, let me make certain i can corral above 500. yup im now at 501.  any thoughts on the bug/enhancement. Keep in mind, even if you do fix the bug, the enhancement could be quite useful.

**Comment by Papa2Cherries** [comment]
small update.  I decided to manually add all the rest and it got up to 774, thinking that perhaps the extension was tring to handle more than 1000.  then i created 4 tabs, 2 windows, and let it count down.  still, it goes to -- and does not wrangle .

**Comment by Papa2Cherries** [comment]
# 9, you can shift click all the links in the tab corral area, but not the tab lock area.  can this be implemented for uniformity as well as functionality.  Makes locking multiple open tabs easier.  Along those lines, perhaps adding a right click on the window "lock all tabs in current window" "unlock all tabs in current window" something like that, so you can have different browser work areas.  something you want to work on the next while, but will ultimately get corralled due to your normal settings.

**Comment by Papa2Cherries** [comment]
if anyone knows why my font is so messed up and how to correct, please respond.

**Comment by ssorallen** [comment]
@Papa2Cherries Thank you for the detailed bugs and features, I really appreciate it.

8. How many tabs do you have set for your max? Currently that number counts *all* tabs Chrome has open across all windows. This is confusing. Issue #95 tracks adding a way for making that count be per-window rather than per-Chrome.

9. Thank you, I forgot I hadn't implemented that for the LockTab. I opened #244 to track that.

**Comment by Papa2Cherries** [comment]
Alright, i see where I went wrong.  What was happening is each window had its own "under 30" limit that was not being reached.  I wish we could turn that on and off, as i see validity reading the linked post you mentioned in having it both ways.  I expected the program to close all but 30 tabs TOTAL, not 30 for each window.  so my confusing #8 can be mostly ignored as "worked as intended."  however, i would like the ability choose one way or the other.  or at the very least, write in the tooltip for max tabs before corral portion that this applies to each window individually.  Sorry for crying wolf, thanks for the quick "fix" in understanding what was going on.

**Comment by Papa2Cherries** [comment]
10, i just noticed today that my tabs went from 742 down to an even 500.  I believe chrome had a crash where i had to do a "restore previously opened tabs" but cannot confirm that that was when this started to happen.  Basically, my settings reset and went from 1000 tab limit to 500 along with the timeout limit for wrangling.  What this resulted in was a much shorter tab wrangle time without me noticing till it was too late.  So i initially lost the 242 tabs since it trimmed it down to 500, then lost even more with the tabs being wrangled at a rate faster than i wanted.  I am uncertain if an update was pushed out recently, which i see that 6.1.5 was posted 22 days ago, so i dont believe it has anything to do with an update.  SO my suggestion is two fold.  one simple solution is to make default the 1000 limit so that way if anyone wants less and their chrome resets the options in anyway, they dont auto trim out a bunch of their tabs right away.  Or two, find out how the options can be reset and save them in a more permanent way.  best i can tell, the only options that were reset were the first three fields that can be written too.  The autolock url area seems to have kept old entries.  so perhaps it has something to do with the number fields and how they are saved.  I dont expect this to be a common occurrence but i literally lost well over 400 tabs that can never be recovered due to this happening.

**Comment by Papa2Cherries** [comment]
I am guessing it was a one time thing.  It has not happened again that i know of.  I do have a separate question, what causes the tabs to start their countdown.  Or moreover, what resets the countdown?  does a computer or browser restart reset the countdown, even if those pages are not viewed?
